File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/setting/SettingRequest.java
package com.portSrilanka.board_admin_backend.dto.setting;

import com.portSrilanka.board_admin_backend.enums.SettingGroup;
import lombok.Builder;
import lombok.Data;

@Data
public class SettingRequest {
    private SettingGroup settingGroup;
    private String settingKey;
    private String settingValue;
    private String description;
}

File: board-admin-frontend/frontend/lib/features/settings/model/setting_request.dart
class SettingRequest {
  final String settingGroup;
  final String settingKey;
  final String settingValue;
  final String? description;

  SettingRequest({
    required this.settingGroup,
    required this.settingKey,
    required this.settingValue,
    this.description,
  });

  Map<String, dynamic> toJson() {
    return {
      'settingGroup': settingGroup,
      'settingKey': settingKey,
      'settingValue': settingValue,
      'description': description,
    };
  }
}
